src/js/messaging/components/MessageAttachmentsViewItem.jsx

```diff
@@ -1,11 +1,52 @@
 import React from 'react';
+import classNames from 'classnames';
+
+import { authFetch } from '../utils/helpers';
 
 class MessageAttachmentsViewItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.downloadAttachment = this.downloadAttachment.bind(this);
+    this.downloadUrl = null;
+    this.state = { downloading: false };
+  }
+
+  downloadAttachment(event) {
+    event.preventDefault();
+
+    if (this.state.downloading) return;
+
+    if (this.downloadUrl) {
+      window.open(this.downloadUrl, '_blank');
+      return;
+    }
+
+    this.setState({ downloading: true });
+    const requestUrl = this.props.url;
+    authFetch(requestUrl)
+    .then(response => response.blob())
+    .then(blob => {
+      const downloadUrl = URL.createObjectURL(blob);
```

Comment:
I don't know of any—the best way to handle large files is to have the API return a signed URL with expiration. Given that the request needs to have an auth header it is impossible to use links, and using AJAX XMLHttpRequest won't trigger a browser file download.

Outside of using signed URLs like S3/Cloudfront do, your approach here is the best way to do it given these limitations